Add README.md file (#97)
unikraft/docs: README.md

> # Unikraft official website.

Reviewer: ```suggestion
# Unikraft Website

This repository stores the Unikraft OSS website, together with documentation, hackathon contents, blog posts and community information.
```

> You can find the documentation under https://unikraft.org/docs/

Reviewer: ```suggestion
Contents from this repository are published at https://unikraft.org/ with sub-links available from the top bar:

- documentation: https://unikraft.org/docs/
- community information (including hackathons): https://unikraft.org/community/
- blog: https://unikraft.org/blog/
```

> 

Reviewer: ```suggestion

The Unikraft website is written in [Markdown](https://www.markdownguide.org/) and published via [Hugo](https://gohugo.io/) and the [Docsy theme](https://www.docsy.dev/).
```

> ## Build the website locally

Reviewer: ```suggestion
## Build
```

You can build the website inside a docker environment.
For this, you will need to install Docker-CE.

> You can do that by following the instructions [here](https://github.com/docker/docker-install), or simply run:

Reviewer: ```suggestion
You can do that by following the instructions [here](https://github.com/docker/docker-install), or simply run:

```

```bash
curl -fsSL https://get.docker.com/ | sh
```

> To then deploy the website, run:

Reviewer: ```suggestion
To then deploy the website, run:

```

```bash
$ make container
$ make devenv

$ make serve # To be run inside the docker env
```

> You may need special privileges to run the first 2 commands.

Reviewer: ```suggestion

You may need special privileges to run the first 2 commands.
```

> You can also deploy the website locally.

Reviewer: ```suggestion
You can also build deploy the website natively.
```

You will need to install `hugo`.

> You can do it by following the instructions [here](https://gohugo.io/getting-started/installing/).

Reviewer: ```suggestion
You can do it by following the instructions [here](https://gohugo.io/getting-started/installing/).

Then, to build and deploy the website, run:

```

```bash
$ hugo serve
```

> Please note that this may lead to errors due to different hugo versions or other environment problems.

Reviewer: ```suggestion

Please note that this may lead to errors due to different hugo versions or other environment problems.
```

> Please note that this may lead to errors due to different hugo versions or other environment problems.

Reviewer: Add:

After building and deploying the website, `hugo` provides instructions on accessing and using it:

```
Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
```

Access the http://localhost:1313/ URL in the browser to view the website.

> # Unikraft Website.

Reviewer: ```suggestion
# Unikraft Website
```

Remove the dot (`.`) ad the end of the title.

Contents from this repository are published at https://unikraft.org/ with sub-links available from the top bar:

- documentation: https://unikraft.org/docs/
- community information (including hackathons): https://unikraft.org/community/
- blog: https://unikraft.org/blog/

> The Unikraft website is written in [Markdown](https://www.markdownguide.org/) and published via [Hugo](https://gohugo.io/) and the [Docsy theme](https://www.docsy.dev/).

Reviewer: ```suggestion
The Unikraft website is written in [Markdown](https://www.markdownguide.org/) and published via [Hugo](https://gohugo.io/) and [the Docsy theme](https://www.docsy.dev/).
```

## Build

> You can build the website inside a docker environment.

Reviewer: ```suggestion
You can build the website inside a Docker environment.
```

```

To then deploy the website, run:

```bash
$ make container
$ make devenv

# To be run inside the docker env
$ npm install
$ make serve
```

You may need special privileges to run the first 2 commands.

You can also build the website natively.
You will need to install `hugo`.
You can do it by following the instructions [here](https://gohugo.io/getting-started/installing/).

Then, to build and deploy the website, run:

```bash
$ hugo serve
```

Please note that this may lead to errors due to different hugo versions or other environment problems.

After building and deploying the website, `hugo` provides instructions on accessing and using it:

```text
Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
```

Access the `http://localhost:1313/` URL in the browser to view the website.

## Tips, Tricks and Best Practices

If you are trying to add or update content on this repository, there are some things you should have in mind.

### Referring pages and sections

You can refer to different pages and section from the [website](https://unikraft.org/) by using paths starting from [`content/en` directory](https://github.com/unikraft/docs/tree/main/content/en).

> For example, say you want to add a reference to the [`development/booting` page](https://unikraft.org/docs/develop/booting/), that is located inside the repo at [this file](https://github.com/unikraft/docs/blob/main/content/en/docs/develop/booting.md), it's location from the repository root beeing `content/en/docs/develop/booting.md`.

Reviewer 2: ```suggestion
For example, say you want to add a reference to the [`development/booting` page](https://unikraft.org/docs/develop/booting/), Inside the repo, the file used to render the page can be found [here](https://github.com/unikraft/docs/blob/main/content/en/docs/develop/booting.md), its location from the repository root being `content/en/docs/develop/booting.md`.
```

> To do this, you can add the following line in your working `.md` page:

Reviewer 2: ```suggestion
To achieve this, you can add the following line in your working `.md` page:
```

You will need to install `hugo`.
You can do it by following the instructions [here](https://gohugo.io/getting-started/installing/).

Then, to build and deploy the website, run:

```bash
$ hugo serve
```

**Please note that this may lead to errors due to different hugo versions or other environment problems.
The problems can be avoided by building the webside inside a Docker environment.
If you choose to build the website natively, make sure you use [`Hugo v0.98.0`](https://github.com/gohugoio/hugo/releases/tag/v0.98.0).**

After building and deploying the website, `hugo` provides instructions on accessing and using it:

```text
Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
```

Access the `http://localhost:1313/` URL in the browser to view the website.

## Tips, Tricks and Best Practices

If you are trying to add or update content on this repository, there are some things you should have in mind.

### Referring pages and sections

You can refer to different pages and section from the [website](https://unikraft.org/) by using paths starting from [`content/en` directory](https://github.com/unikraft/docs/tree/main/content/en).

> For example, say you want to add a reference to the [`development/booting` page](https://unikraft.org/docs/develop/booting/). Inside the reposotory, the file used to render the page can be found [here](https://github.com/unikraft/docs/blob/main/content/en/docs/develop/booting.md), its location from the repository root being `content/en/docs/develop/booting.md`.

Reviewer 2: ```suggestion

For example, say you want to add a reference to the [`development/booting` page](https://unikraft.org/docs/develop/booting/). Inside the repository, the file used to render the page can be found [here](https://github.com/unikraft/docs/blob/main/content/en/docs/develop/booting.md), its location from the repository root being `content/en/docs/develop/booting.md`.
```

> To achieve this, you can add the following line in your working `.md` page:

Reviewer 2: ```suggestion

To achieve this, you can add the following line in your working `.md` page:
```

```
[your reference](docs/develop/booting)
```

givind the path of the file starting after the `content/en` directory.

You can also refer specific sections of one page by adding `/#section` to the end of the path.
You can find the sections of a page listed on the rigth side of the page.

> For example, a reference to the [`ARM` section of the booting page](https://unikraft.org/docs/develop/booting/#arm) would look like this:

Reviewer 2: ```suggestion

For example, a reference to the [`ARM` section of the booting page](https://unikraft.org/docs/develop/booting/#arm) would look like this:
```

# Unikraft Website

Contents from this repository are published at https://unikraft.org/ with sub-links available from the top bar:

- documentation: https://unikraft.org/docs/
- community information (including hackathons): https://unikraft.org/community/
- blog: https://unikraft.org/blog/

The Unikraft website is written in [Markdown](https://www.markdownguide.org/) and published via [Hugo](https://gohugo.io/) and the [Docsy theme](https://www.docsy.dev/).

## Build

You can build the website inside a Docker environment.
For this, you will need to install Docker-CE.

> You can do that by following the instructions [here](https://github.com/docker/docker-install), or simply run:

Reviewer: ```suggestion
You can do that by following the instructions [here](https://github.com/docker/docker-install), or by running:
```

> **Note: This is highly recomended, since you will work in the same environment as the deployed website, so there will be no errors regarding packages version or filesystem layout.**

Reviewer 2: ```suggestion
**Note: This is highly recommended, since you will work in the same environment as the deployed website, so there will be no errors regarding packages version or filesystem layout.**
```

> **Note: This is highly recommended, since you will work in the same environment as the deployed website, so there will be no errors regarding packages version or filesystem layout.**

Reviewer: What is "This"? Replace "This" with the actual item.

```bash
curl -fsSL https://get.docker.com/ | sh
```

To then deploy the website, run:

```bash
$ make container
$ make devenv

# To be run inside the docker env
$ npm install
$ make serve
```

> You may need special privileges to run the first 2 commands.

Reviewer: ```suggestion
You may need admin privileges to run the first 2 commands.
```

You can also build the website natively.

> You will need to install `hugo`.

Reviewer: ```suggestion
For a native build, you need to install `hugo`.
```

You can do it by following the instructions [here](https://gohugo.io/getting-started/installing/).

Then, to build and deploy the website, run:

```bash
$ hugo serve
```

> **Please note that this may lead to errors due to different hugo versions or other environment problems.

Reviewer 2: ```suggestion
**Note: This may lead to errors due to different `hugo` versions or other environment problems.
```
Changed this in order to match the other "Note:" that we have in the document.

The problems can be avoided by building the webside inside a Docker environment.
If you choose to build the website natively, make sure you use [`Hugo v0.98.0`](https://github.com/gohugoio/hugo/releases/tag/v0.98.0).**

After building and deploying the website, `hugo` provides instructions on accessing and using it:

```text
Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
```

Access the `http://localhost:1313/` URL in the browser to view the website.

## Tips, Tricks and Best Practices

If you are trying to add or update content on this repository, there are some things you should have in mind.

### Referring pages and sections

> You can refer to different pages and section from the [website](https://unikraft.org/) by using paths starting from [`content/en` directory](https://github.com/unikraft/docs/tree/main/content/en).

Reviewer: ```suggestion
You can refer to different pages and sections ine the [website](https://unikraft.org/) by using paths starting from [`content/en` directory](https://github.com/unikraft/docs/tree/main/content/en).
```
